### DISCUSSION TITLE
Includes "deepvar" and "gpvar" models in GluonTS.

### DISCUSSION THREAD

**Discussion by mreismendes** [winedarksea/AutoTS#24] [Ideas]
Hi!

Could someone include DeepVar and GPVar models in GluonTS?

What do you think about that?

Thank you.

**Comment by winedarksea** [comment]
These are both models added to GluonTS after I built the initial implementation here...  I am happy to include them (at some point, soon, my current concern is revamping some of the transformers). 
I see it says that DeepVar is "multivariate" - I am not sure if that affects the implementation at all. I don't suppose you know anything more about those new models, any tips or tricks I should keep in mind if I add them in?

**Reply by winedarksea** [reply]
Example found here: https://github.com/awslabs/gluon-ts/issues/1095

**Reply by mreismendes** [reply]
I did some experiments with them, Gluonts docs are very confusing at least to me! But after long research, I discovered I should group the targets.

![image](https://user-images.githubusercontent.com/32521301/101837920-916f9a00-3b1e-11eb-9e35-131699faa4b1.png)
 
And then, training!
![image](https://user-images.githubusercontent.com/32521301/101838119-d267ae80-3b1e-11eb-92fc-204d8c5a257f.png)

**Reply by mreismendes** [reply]
@winedarksea I think your example it's a lot better than mine!

**Reply by winedarksea** [reply]
Thanks for the examples, that will help!

**Reply by mreismendes** [reply]
Usually, I work with commodities time-series very often. In those data, I noticed that when you apply at least a year of context improves a lot the validation metrics. Usually, those commodities here have a cycle of one year, like cattle!

I don't have a theoretical explanation of the reason!

Example 01: 
```
freq='M',
prediction_length=3
prediction_length=12
```

Example 02: 
```
freq='W',
prediction_length=10
prediction_length=52

```
The drawback is that usually takes a lot more time to train!
